tests/test_diagnostics.py

```
@@ -0,0 +1,311 @@
+            )
+
+    def test_empty_agent_message_rejected(self):
+        with pytest.raises(ValueError, match="agent_message must be non-empty"):
+            TaxDiagnosticResult(
+                status=TaxDiagnosticStatus.BLOCKED,
+                agent_message="",
+            )
+
+    def test_frozen_dataclass(self):
+        result = TaxDiagnosticResult.blocked("blocked")
+        with pytest.raises(Exception):
```

Ruff (0.15.17)

[warning] Do not assert blind exception: `Exception`

(B017)